pkg/provider/provider_test.go

```diff
@@ -862,3 +862,72 @@ func Test_buildContainerImageSource(t *testing.T) {
 		})
 	}
 }
+
+func TestTestEnvironment_GetBaremetalNodes(t *testing.T) {
+	type fields struct {
+		Nodes map[string]Node
```

Comment:
Why use a `map`?  Seems like you would just need a `[]Node`.

Reply:
Then just compare the slices.  Actually in Go 1.21 they added a `slices` comparison tool which we could take advantage of after we move to 1.21.

Reply:
i need it as that type since that what we are using in env